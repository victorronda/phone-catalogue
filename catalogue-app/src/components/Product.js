import React from 'react';
import { Link } from 'react-router-dom';

const Product = ({ phone }) => {

    
	return (
		<div>
			<Link to={`/details/${phone.id}`} phone={phone}>
				<img src={phone.imageFileName} alt="phone" />
				<h2>{phone.name}</h2>
				<h3>{phone.price} â‚¬</h3>
			</Link>
		</div>
	);
};



export default Product;
