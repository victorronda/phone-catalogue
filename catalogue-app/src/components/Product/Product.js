import React from 'react';
import { Link } from 'react-router-dom';

const Product = ({ phone }) => {
	return (
		<div className="card m-4 text-center rounded-lg shadow-lg" style={{width: "15em"}}>
			<Link className="text-decoration-none text-dark" to={`/details/${phone.id}`} phone={phone}>
				<img className="card-img-top" src={phone.imageFileName} alt="phone" width="auto"
                    height="250"/>
				<div className="card-body">
					<h2 className="card-title">{phone.name}</h2>
					<h3 lassName="card-subtitle">{phone.price} â‚¬</h3>
				</div>
			</Link>
		</div>
	);
};

export default Product;
