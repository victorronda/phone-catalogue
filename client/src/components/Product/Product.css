.hover-card:hover {
    transform: scale(1.03);
    transition: all 100ms ease-in;
    transition-delay: 50ms;
}

