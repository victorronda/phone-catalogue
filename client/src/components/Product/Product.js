import React from 'react';
import { Link } from 'react-router-dom';
import './Product.css';

const Product = ({ phone }) => {
	return (
		<div className="card m-4 text-center rounded-lg shadow-lg hover-card" style={{ width: '16em' }}>
			<Link className="text-decoration-none text-dark" to={`/details/${phone.id}`} phone={phone}>
				<img className="card-img-top" src={phone.imageFileName} alt="phone" width="auto" height="280" />
				<div className="card-body">
					<h2 className="card-title">{phone.name}</h2>
					<h3 className="card-subtitle">{phone.price} â‚¬</h3>
				</div>
			</Link>
		</div>
	);
};

export default Product;
