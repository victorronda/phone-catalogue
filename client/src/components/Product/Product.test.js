import { shallow } from 'enzyme';
import React from 'react';
import Product from './Product';

it('should render Product component', () => {
    const mockPhone = [
        {
            id: 5,
            name: 'Nokia',
            ram: 2,
            imageFileName: 'http://localhost:4000/images/IPhone_7.png'
        }
    ];
    expect(shallow(<Product phone={mockPhone} />)).toMatchSnapshot();
});