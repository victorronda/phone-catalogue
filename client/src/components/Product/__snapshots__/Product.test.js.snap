// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Product component 1`] = `ShallowWrapper {}`;
